xxhash.h

```diff
@@ -291,7 +291,7 @@ struct XXH64_state_s {
    uint64_t v4;
    uint64_t mem64[4];
    uint32_t memsize;
-   uint32_t reserved[2];   /* never read nor write, might be removed in a future version */
+   uint32_t reserved;   /* never read nor write, might be removed in a future version */
```

Comment:
Why don't we just remove reserved entirely? It *is* a future version…

Reply:
I don't want to introduce risks of ABI break while also introducing `xxh3`.

I believe `uint32_t reserved[2];` can or even should be replaced by `uint64_t reserved;`. It doesn't change anything regarding the structure size, it would just be "proper".

`reserved` might be removed in a future version, or it might remain there forever.
The tiny memory savings pale in comparison of risks of ABI break.

Reply:
the question is "who's responsible for not crashing inconsistent headers and libs?", because it has some historical footprint